File: exercises_00/03.py
"""
    Program that:
        Escriba un programa que calcule el área total del cuerpo
"""

import math

def main () -> None:
    weight: float = 0
    height: float = 0

    try:
        height = float(input())
        weight = float(input())

    except ValueError as e:
        print(f"ERROR VALOR INVALIDO:\n{type(e)}\n{e}")
        return
    except Exception as e:
        print(f"ERROR:\n{type(e)}\n{e}")
        return

    result_mosteller: float = calc_body_surface_area_mosteller(weight, height)
    result_dubois: float = calc_body_surface_area_dubois(weight, height)
    # result_haycock: float = calc_body_surface_area_haycock(weight, height)
    result_gehan_george: float = calc_body_surface_area_gehan_george(weight, height)

    print(f"{result_mosteller:.2f}")
    print(f"{result_dubois:.2f}")
    # print(f"{result_haycock:.2f}")
    print(f"{result_gehan_george:.2f}")

def calc_body_surface_area_mosteller (weight: float, height: float) -> float:
    return math.sqrt((weight * height) / 3600)

def calc_body_surface_area_haycock (weight: float, height: float) -> float:
    return (0.024265) * (weight ** 0.5378) * (height ** 0.3964)

def calc_body_surface_area_dubois (weight: float, height: float) -> float:
    return 0.007184 * (weight ** 0.425) * (height ** 0.725)

def calc_body_surface_area_gehan_george (weight: float, height: float) -> float:
    return 0.0235 * (height ** 0.42246) * (weight ** 0.51456)

if __name__ == '__main__':
    main()

File: c_exercises_00/03.c
/*
 * =====================================================================================
 *
 *       Filename:  03.c
 *
 *    Description: Escriba un programa que calcule el área total del cuerpo 
 *
 *        Version:  1.0
 *        Created:  08/30/2023 05:49:13 PM
 *       Compiler:  gcc
 *
 * =====================================================================================
 */

#include <stdio.h>
#include <math.h>

int main (void) {
  double weight = 0.0;
  double height = 0.0;

  double resultMosteller = 0.0;
  double resultDubois = 0.0;
  double resultGehanGeorge = 0.0;
  
  scanf("%lf", &height);
  scanf("%lf", &weight);

  resultMosteller = pow((weight * height) / 3600.0, 1.0 / 2);
  resultDubois = 0.007184 * pow(weight, 0.425) * pow(height, 0.725);
  resultGehanGeorge = (0.0235) * pow(height, 0.42246) * pow(weight, 0.51456);

  printf("%.2lf\n%.2lf\n%.2lf", resultMosteller, resultDubois, resultGehanGeorge);

  return 0;
}
